Objects/bytearrayobject.c

```diff
@@ -184,7 +184,12 @@ PyByteArray_Resize(PyObject *self, Py_ssize_t requested_size)
     assert(self != NULL);
     assert(PyByteArray_Check(self));
     assert(logical_offset <= alloc);
-    assert(requested_size >= 0);
+
+    if (requested_size < 0) {
+        PyErr_Format(PyExc_BufferError,
+            "Can only resize to positive sizes, got %zd.", requested_size);
```

Comment:
IMO ValueError is more appropriate. And Python error messages usually don't end with a dot.

```suggestion
        PyErr_Format(PyExc_ValueError,
            "Can only resize to positive sizes, got %zd", requested_size);
```

Reply:
BufferError is what existing bytearray resize failures raise: https://github.com/python/cpython/blob/main/Objects/bytearrayobject.c#L70-L79 (`_canresize`). I think the `bytearray` internal consistency is more useful than returning ValueError.

Will remove the period

Reply:
actually, just found construction does `ValueError` for negative length, updating:
```
    def test_constructor_value_errors(self):
        self.assertRaises(ValueError, self.type2test, [-1])
```